projects/core/src/main/java/dan200/computercraft/core/filesystem/FileMount.java

```diff
@@ -143,6 +143,21 @@ public boolean isDirectory(String path) {
         return file.exists() && file.isDirectory();
     }
 
+    @Override
+    public boolean isReadOnly(String path) throws IOException {
+        var file = getRealPath(path);
+        if (file.exists()) {
+            return !file.canWrite();
+        }
+
+        var parent = file.getParentFile();
+        while (!parent.exists() && parent != rootPath) {
+            parent = parent.getParentFile();
+        }
+
+        return !parent.canWrite();
```

Comment:
I wonder if it's better to write this something like:

```java
while (true) {
  if (file.exists()) return !file.canWrite();
  if (file.equals(rootPath)) return false;
  file = file.getParentFile();
}
```

I think this is the same, except it means `isReadOnly` will return `false` when the root doesn't exist?

Actually, I wonder if it's worth adding some tests for these? Happy to do so myself if you'd rather not!

Reply:
![image](https://user-images.githubusercontent.com/858456/203850133-54c864d4-91f0-4b21-a30a-7ab4465f8475.png)

Seems to work! Thanks, that's much more concise.